data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/PeerForwarder.java

```diff
@@ -85,7 +96,9 @@ public PeerForwarder(final PluginSetting pluginSetting, final PeerForwarderConfi
         }
 
         // Buffer of requests to be exported to the downstream of the local data-prepper
-        final List<Record<ExportTraceServiceRequest>> results = new ArrayList<>();
+        final List<Record<ExportTraceServiceRequest>> results = new LinkedList<>();
```

Comment:
It's  _probably_ an unnecessary change, but because we're never accessing elements by index I don't see a reason to maintain an ArrayList (which has to resize after every so many additions).